### PR TITLE
Protect post access

### PR DESCRIPTION
**Pull request by giancarlopernudisegura**
Remove private posts from appearing in the API results and disallow private posts from being edited according to the user stories.

### CODE REVIEW

File: server/routes/api.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def single_author(author_id: int) -> Response:
 
 @bp.route("/authors/<int:author_id>/posts/<int:post_id>", methods=["GET"])
 def get_post(author_id: int, post_id: int) -> Response:
```
Comment (dalidia) [inline server/routes/api.py:85]: should we restrict the access to this post for only users logged in?

Comment (giancarlopernudisegura) [reply server/routes/api.py:85]: It's only returning public posts so I don't believe we need authentication right now.

Comment (dalidia) [reply server/routes/api.py:85]: makes sense

**Comment by Pegmode** [comment]
I don't think the requirements say anything about not allowing someone to edit their own private posts.
The only mention of editing a post in the requirements is: *"As an author I want to edit public posts."*

The user story on which this is based https://github.com/giancarlopernudisegura/cmput404/issues/31 says:  

*"As an author, **other authors** cannot modify my post shared to friends.."* 

This doesn't say anything about the author who made the post. 

Also the dis-inclusion of private posts in the user story *"As an author I want to edit public posts."* does not imply that only public posts can be edited in my view. 

I suggest that disallowing private post editing is unnecessary and the only check that needs to be done for editing a post should be if the current user is authenticated and the posts author.

**Review by Pegmode — CHANGES_REQUESTED** [review]
I suggest removing this https://github.com/giancarlopernudisegura/cmput404/pull/85/commits/3a65b869fd2c8d02b4e7b4e235642147c73ce6f1 commit as per https://github.com/giancarlopernudisegura/cmput404/pull/85#issuecomment-1050382390

**Review by Pegmode — APPROVED** [review]
Changes look good

**Pull request merged**